### PR TITLE
Fix create exe tests

### LINKED ISSUE #3481: Create-exe with Serialized object fail
### Describe the bug

CI tests `create_exe_serialized_works`  and `create_exe_with_object_input_serialized` have been disable because they are not working.

The generated EXE is somehow incorect, and just segfault during some Rust's Stdlib initialization.

Isolating a binary and running it with gdb gives this output:

````
─── Output/messages ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7d1eff3 in __GI_getenv (name=0x16b0810 "RUST_BACKTRACE") at ./stdlib/getenv.c:51
51      ./stdlib/getenv.c: No such file or directory.
─── Assembly ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 0x00007ffff7d1efe1  __GI_getenv+33 movzbl (%rdi),%eax
 0x00007ffff7d1efe4  __GI_getenv+36 mov    %rdi,%r13
 0x00007ffff7d1efe7  __GI_getenv+39 test   %al,%al
 0x00007ffff7d1efe9  __GI_getenv+41 je     0x7ffff7d1f088 <__GI_getenv+200>
 0x00007ffff7d1efef  __GI_getenv+47 cmpb   $0x0,0x1(%rdi)
 0x00007ffff7d1eff3  __GI_getenv+51 mov    0x0(%rbp),%rbx
 0x00007ffff7d1eff7  __GI_getenv+55 jne    0x7ffff7d1f030 <__GI_getenv+112>
 0x00007ffff7d1eff9  __GI_getenv+57 or     $0x3d,%ah
 0x00007ffff7d1effc  __GI_getenv+60 test   %rbx,%rbx
 0x00007ffff7d1efff  __GI_getenv+63 jne    0x7ffff7d1f015 <__GI_getenv+85>
─── Breakpoints ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
─── Expressions ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
─── History ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
─── Memory ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
─── Registers ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
    rax 0x0000000000000052    rbx 0x0000000000000000    rcx 0x0000000000000001    rdx 0x000000000000000f    rsi 0x000000000000000f       rdi 0x00000000016b0810    rbp 0x00000000016d1000    rsp 0x00007fffffffb440    r8 0x00007ffff7eb2c60    r9 0x0000000000000000    r10 0x0000000000000022    r11 0x0000000000000020
    r12 0x0000000000000001    r13 0x00000000016b0810    r14 0x00007fffffffb4e8    r15 0x00007fffffffb680    rip 0x00007ffff7d1eff3    eflags [ PF IF RF ]           cs 0x00000033             ss 0x0000002b            ds 0x00000000            es 0x00000000             fs 0x00000000             gs 0x00000000
─── Source ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Cannot display "getenv.c"
─── Stack ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
[0] from 0x00007ffff7d1eff3 in __GI_getenv+51 at ./stdlib/getenv.c:51
[1] from 0x000000000038ec50 in std::sys::unix::os::getenv+141 at library/std/src/sys/unix/os.rs:552
[2] from 0x000000000038ec50 in std::env::_var_os+160 at library/std/src/env.rs:273
[3] from 0x000000000039548f in std::env::var_os<&str>+23 at library/std/src/env.rs:269
[4] from 0x000000000039548f in std::panic::get_backtrace_style+63 at library/std/src/panic.rs:303
[5] from 0x0000000000399226 in std::panicking::default_hook+54 at library/std/src/panicking.rs:269
[6] from 0x0000000000399d36 in std::panicking::rust_panic_with_hook+246 at library/std/src/panicking.rs:698
[7] from 0x0000000000399be9 in std::panicking::begin_panic_handler::{closure#0}+121 at library/std/src/panicking.rs:586
[8] from 0x0000000000398454 in std::sys_common::backtrace::__rust_end_short_backtrace<std::panicking::begin_panic_handler::{closure_env#0}, !>+20 at library/std/src/sys_common/backtrace.rs:138
[9] from 0x0000000000399959 in std::panicking::begin_panic_handler+73 at library/std/src/panicking.rs:584
[+]
─── Threads ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
[1] id 2683 name wasm.out from 0x00007ffff7d1eff3 in __GI_getenv+51 at ./stdlib/getenv.c:51
─── Variables ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
arg name = 0x16b0810 "RUST_BACKTRACE": 82 'R'
loc ep = <optimized out>, name_start = <optimized out>
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
>>> x/x $rbp
0x16d1000:      Cannot access memory at address 0x16d1000
>>> q
```

### PR DESCRIPTION
**Pull request by fschutt**
Fixes #3481.

### CODE REVIEW

File: lib/compiler/src/artifact_builders/artifact_builder.rs [hunk 1]
```diff
@@ -119,17 +119,10 @@ impl ArtifactBuild {
             compile_info,
             data_initializers,
             cpu_features: cpu_features.as_u64(),
-            module_start: None,
         };
         Ok(Self { serializable })
     }
 
-    /// Specify the fixed virtual memory address for the compiled module
```
Comment (theduke) [inline lib/compiler/src/artifact_builders/artifact_builder.rs:127]: You can't just remove the whole module_start stuf...

We might not need it anymore, but that needs to be determined with John.

Comment (fschutt) [reply lib/compiler/src/artifact_builders/artifact_builder.rs:127]: Yeah that's what I wanted to sync on.

Comment (fschutt) [reply lib/compiler/src/artifact_builders/artifact_builder.rs:127]: I just wanted to show that it fixes the issue in the CI.

Comment (theduke) [reply lib/compiler/src/artifact_builders/artifact_builder.rs:127]: @fschutt good news, we actually can remove it!

File: lib/compiler-llvm/src/abi/aarch64_systemv.rs [hunk 2]
```diff
@@ -21,19 +21,14 @@ impl Abi for Aarch64SystemV {
     // Given a function definition, retrieve the parameter that is the vmctx pointer.
     fn get_vmctx_ptr_param<'ctx>(&self, func_value: &FunctionValue<'ctx>) -> PointerValue<'ctx> {
         func_value
-            .get_nth_param(
```
Comment (theduke) [inline lib/compiler-llvm/src/abi/aarch64_systemv.rs:24]: I assume this is it a clippy lint?

The old code was much more readable, should use allow instead.

File: lib/compiler-llvm/src/abi/x86_64_systemv.rs [hunk 3]
```diff
@@ -23,19 +23,14 @@ impl Abi for X86_64SystemV {
     // Given a function definition, retrieve the parameter that is the vmctx pointer.
     fn get_vmctx_ptr_param<'ctx>(&self, func_value: &FunctionValue<'ctx>) -> PointerValue<'ctx> {
         func_value
-            .get_nth_param(
-                if func_value
```
Comment (theduke) [inline lib/compiler-llvm/src/abi/x86_64_systemv.rs:27]: as above

File: lib/compiler-llvm/src/translator/code.rs [hunk 4]
```diff
@@ -1048,6 +1048,7 @@ impl<'ctx, 'a> LLVMFunctionCodeGenerator<'ctx, 'a> {
         Ok(())
     }
 
+    #[allow(clippy::unnecessary_cast)]
```
Comment (theduke) [inline lib/compiler-llvm/src/translator/code.rs:1051]: Why? 

Remove the cast instead?

Comment (theduke) [reply lib/compiler-llvm/src/translator/code.rs:1051]: Same goes for all the others in this file.

**Review by theduke — CHANGES_REQUESTED** [review]
@fschutt why were the clippy issues not caught on CI before the CI refactor merge?

File: lib/compiler-llvm/src/translator/code.rs (hunk 4 above)
Comment (fschutt) [reply lib/compiler-llvm/src/translator/code.rs:1051]: @theduke If I remove the cast, there might be unexpected issues on 32-bit vs 64-bit platforms, that's why I ignored the lint instead.

File: lib/compiler/src/artifact_builders/artifact_builder.rs (hunk 1 above)
Comment (fschutt) [reply lib/compiler/src/artifact_builders/artifact_builder.rs:127]: Okay, then I can leave it as is then.

**Comment by fschutt** [comment]
> @fschutt why were the clippy issues not caught on CI before the CI refactor merge?

Yes, this is something we need to sync on. For some reason I always see a green master, then I make my changes and then suddenly make lint complains and is full of bugs, which makes subsequent PRs harder (I did not expect make lint to fail).

For some reason the CI doesn't properly catch make lint failures, I opened https://github.com/wasmerio/wasmer/issues/3489 to track this.

File: lib/compiler-llvm/src/abi/aarch64_systemv.rs (hunk 2 above)
Comment (fschutt) [reply lib/compiler-llvm/src/abi/aarch64_systemv.rs:24]: Done in https://github.com/wasmerio/wasmer/pull/3487/commits/ddc7628b7f6fd97318abb400eed0d1ecb6a78990

File: lib/compiler-llvm/src/translator/code.rs (hunk 4 above)
Comment (fschutt) [reply lib/compiler-llvm/src/translator/code.rs]: I've removed it now with the lints that were working.

**Review by syrusakbary — APPROVED** [review]

**Comment by fschutt** [comment]
Done in #3496.

**Pull request closed without merging**